src/ArraySet/ArraySet.spec.ts

```diff
@@ -0,0 +1,267 @@
+                    )
+                ).toStrictEqual(["A", "C"])
+            })
+
+            it("does not remove a non-extant element from the set", () => {
+                expect(
+                    pipe(
+                        ArraySet.of(["A", "B", "C"]),
+                        ArraySet.remove("D", caseInsensitiveEqualityComparer)
+                    )
+                ).toStrictEqual(["A", "B", "C"])
+            })
+        })
+    })
+
+    describe("ofArray", () => {
+        it("returns an array with duplicates removed (default equality)", () => {
+            expect(ArraySet.of([1, 3, 1, 3, 5, 6, 7, 4, 4])).toStrictEqual([
+                1, 3, 5, 6, 7, 4,
+            ])
+        })
+
+        it("returns an array with duplicates removed (custom equality)", () => {
+            expect(
+                ArraySet.ofArray(
+                    ["A", "D", "d", "A", "E", "f", "F"],
+                    caseInsensitiveEqualityComparer
+                )
+            ).toStrictEqual(["A", "D", "E", "f"])
+        })
+    })
+
+    describe("union", () => {
+        describe("default equality", () => {
+            it("produces a set union", () => {
+                expect(
+                    pipe(
+                        ArraySet.of([1, 2, 3]),
+                        ArraySet.union(ArraySet.of([2, 3, 4]))
+                    )
+                ).toStrictEqual([1, 2, 3, 4])
+            })
+        })
+
+        describe("custom equality", () => {
+            it("produces a set union", () => {
+                expect(
+                    pipe(
+                        ArraySet.of(["A", "B", "C", "D"]),
+                        ArraySet.union(
+                            ArraySet.of(["c", "d", "e"]),
+                            caseInsensitiveEqualityComparer
+                        )
+                    )
+                ).toStrictEqual(["A", "B", "C", "D", "e"])
+            })
+        })
+    })
+
+    describe("intersect", () => {
+        describe("default equality", () => {
+            it("produces a set intersection", () => {
+                expect(
+                    pipe(
+                        ArraySet.of([1, 2, 3, 4, 5]),
+                        ArraySet.intersect(ArraySet.of([2, 4, 6]))
+                    )
+                ).toStrictEqual([2, 4])
+            })
+        })
+
+        describe("custom equality", () => {
+            it("produces a set intersection (1)", () => {
+                expect(
+                    pipe(
+                        ArraySet.of(["A", "b", "C", "d", "E"]),
+                        ArraySet.intersect(
+                            ArraySet.of(["c", "e", "B", "X"]),
+                            caseInsensitiveEqualityComparer
+                        )
+                    )
+                ).toStrictEqual(["b", "C", "E"])
+            })
+
+            it("produces a set intersection (2)", () => {
```

Comment:
⛏️ Is the idea here testing which array is longer? It's not necessarily clear from the `(1)` and `(2)` in the test titles.

Reply:
Yea, just testing the behavior in both cases (longer vs shorter).. i'll use some more helpful names than `(1)` and `(2)` 😅

Reply:
https://github.com/fp-toolkit/fp-toolkit/pull/67/commits/f44500c18ba97d3e1a597457e851ad69cce80ff6